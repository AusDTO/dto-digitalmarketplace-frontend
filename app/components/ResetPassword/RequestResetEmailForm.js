import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import PageAlert from '@gov.au/page-alerts'
import { required, validEmail } from '../validators'

export const RequestResetEmailForm = ({ model, form, user, submitClicked, handleSubmit }) => {
  let { resetPasswordEmailFailure, resetPasswordEmailSuccess } = user
  let { valid } = form

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {resetPasswordEmailFailure &&
            valid &&
            <PageAlert as="error">
              <h4>We were unable to send your password reset email</h4>
              <span>
                <p>
                  Please try again later or{' '}
                  <a href="/contact-us" target="_blank" rel="external">
                    {' '}contact us{' '}
                  </a>{' '}
                  for assistance.
                </p>
              </span>
            </PageAlert>}
          {resetPasswordEmailSuccess &&
            <PageAlert as="success">
              <span>
                If the email address you&#39;ve entered belongs to a Marketplace account, we&#39;ll send a link to reset
                the password.
              </span>
            </PageAlert>}
          <ErrorBox model={model} submitClicked={submitClicked} setFocus={setFocus} />
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="uikit-display-5">Reset password</h1>
            <span>
              Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
              valid for 24 hours.
            </span>
          </header>
          <Form model={model} id="sendResetEmail" onSubmit={model => handleSubmit(model)}>
            <Textfield
              model={`${model}.email_address`}
              name="email_address"
              id="email_address"
              htmlFor="email_address"
              description="Enter the email address you used to register with the Marketplace"
              validators={{ required, validEmail }}
              messages={{
                required: 'Your email is required',
                validEmail: 'A valid email is required.'
              }}
            />
            <p>
              <input className="uikit-btn" type="submit" value="Send reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}
