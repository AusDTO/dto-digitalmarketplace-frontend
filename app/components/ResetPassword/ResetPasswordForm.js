import React from 'react'
import { Form } from 'react-redux-form'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import PageAlert from '@gov.au/page-alerts'
import { passwordLength } from '../validators'
import { rootPath } from '../../routes'

export const ResetPasswordForm = ({ model, user, submitClicked, handleSubmit }) => {
  let { resetPasswordSuccess, getResetDataSuccess, errorMessage } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {(resetPasswordSuccess === false || getResetDataSuccess === false) &&
            <PageAlert as="error">
              <h4>We are unable to reset your password</h4>
              <span>
                {errorMessage}
                {getResetDataSuccess === false &&
                  <span>
                    Try <a href={`${rootPath}/reset-password`}> resending </a>your reset password email
                  </span>}
              </span>
            </PageAlert>}
          {resetPasswordSuccess &&
            <PageAlert as="success">
              <h4>You have successfully changed your password</h4>
              <span>
                <p>
                  Please <a href="/login"> login </a> to continue.
                </p>
              </span>
            </PageAlert>}
          <ErrorBox model={model} submitClicked={submitClicked} setFocus={setFocus} />
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="uikit-display-5">Reset password</h1>
          </header>
          <Form
            model={model}
            id="resetPassword"
            validators={{
              '': {
                // Form-level validator
                passwordsMatch: val => passwordsMatch(val)
              }
            }}
            onSubmit={model => handleSubmit(model)}
          >
            <Textfield
              model={`${model}.password`}
              name="password"
              id="password"
              htmlFor="password"
              label="Password"
              type="password"
              description="Your password should be at least 10 characters"
              validators={{ passwordLength }}
              messages={{
                passwordLength: 'Your password should be at least 10 characters'
              }}
            />

            <Textfield
              model={`${model}.confirmPassword`}
              name="confirmPassword"
              id="confirmPassword"
              htmlFor="confirmPassword"
              label="Password"
              type="password"
              description="Repeat password used above"
              validators={{
                passwordLength
              }}
              messages={{
                passwordLength: 'Your password confirmation should be at least 10 characters. ',
                passwordsMatch: 'Passwords do not match.'
              }}
            />
            <p>
              <input className="uikit-btn" type="submit" value="Reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}
