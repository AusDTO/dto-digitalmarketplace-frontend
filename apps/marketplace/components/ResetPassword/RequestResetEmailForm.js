import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from 'shared/form/ErrorBox'
import Textfield from 'shared/form/Textfield'
import { required, validEmail } from '../validators'

const RequestResetEmailForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordEmailSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {resetPasswordEmailSuccess
            ? <AUpageAlert as="success">
                <span>
                  We have sent a link to reset the password.<br />
                  Check your spam folder if it does not arrive.
                </span>
              </AUpageAlert>
            : <ErrorBox
                title="There was a problem sending your reset email"
                model={model}
                submitClicked={submitClicked}
                setFocus={setFocus}
              />}
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="au-display-xl">Reset password</h1>
            <p>
              Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
              valid for 24 hours.
            </p>
          </header>
          <Form model={model} id="sendResetEmail" onSubmit={data => handleSubmit(data)}>
            <Textfield
              model={`${model}.email_address`}
              name="email_address"
              id="email_address"
              htmlFor="email_address"
              description="Enter the email address you used to register with the Marketplace"
              validators={{ required, validEmail }}
              messages={{
                required: 'Your email is required',
                validEmail: 'A valid email is required.'
              }}
            />
            <p>
              <input className="au-btn" type="submit" value="Send reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}

RequestResetEmailForm.propTypes = {
  model: PropTypes.string.isRequired,
  user: PropTypes.shape({
    resetPasswordEmailSuccess: PropTypes.bool
  }).isRequired,
  submitClicked: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default RequestResetEmailForm
