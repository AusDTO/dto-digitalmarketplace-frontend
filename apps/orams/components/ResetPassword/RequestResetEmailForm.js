import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from '../../../shared/form/ErrorBox'
import Textfield from '../../../shared/form/Textfield'
import { required, validEmail } from '../../../shared/validators'

import styles from './ResetPassword.scss'

const RequestResetEmailForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordEmailSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <main>
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {resetPasswordEmailSuccess
              ? <AUpageAlert as="success">
                  <span>
                    If the email address you&#39;ve entered belongs to an ORAMS account, we&#39;ll send a link to reset
                    the password. Check your spam folder if it does not arrive.
                  </span>
                </AUpageAlert>
              : <ErrorBox
                  title="There was a problem sending your reset email"
                  model={model}
                  submitClicked={submitClicked}
                  setFocus={setFocus}
                />}
            <header className="page-heading page-heading-without-breadcrumb">
              <h1 className="au-display-xl">Reset password</h1>
              <span>
                Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
                valid for 24 hours.
              </span>
            </header>
            <Form model={model} id="sendResetEmail" onSubmit={data => handleSubmit(data)}>
              <Textfield
                model={`${model}.email_address`}
                name="email_address"
                id="email_address"
                htmlFor="email_address"
                label="Enter your email address registered to the ORAMS Portal"
                validators={{ required, validEmail }}
                messages={{
                  required: 'Your email is required',
                  validEmail: 'A valid email is required.'
                }}
              />
              <p className={styles.buttonWrapper}>
                <input className="au-btn" type="submit" value="Send reset password" onClick={submitClicked} />
              </p>
            </Form>
          </article>
        </div>
      </div>
    </main>
  )
}

RequestResetEmailForm.propTypes = {
  model: PropTypes.string.isRequired,
  user: PropTypes.shape({
    resetPasswordEmailSuccess: PropTypes.bool
  }).isRequired,
  submitClicked: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default RequestResetEmailForm
