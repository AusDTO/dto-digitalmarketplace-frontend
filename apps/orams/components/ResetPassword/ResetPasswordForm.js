import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from '../../../shared/form/ErrorBox'
import Textfield from '../../../shared/form/Textfield'
import { passwordLength } from '../../../shared/validators'
import { rootPath } from '../../routes'

import styles from './ResetPassword.scss'

const ResetPasswordForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <main>
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {resetPasswordSuccess
              ? <AUpageAlert as="success">
                  <h4>You have successfully changed your password</h4>
                  <span>
                    <p>
                      Please <a href={`${rootPath}/login`}> login </a> to continue.
                    </p>
                  </span>
                </AUpageAlert>
              : <ErrorBox
                  title="There was a problem resetting your password"
                  model={model}
                  submitClicked={submitClicked}
                  setFocus={setFocus}
                />}
            <header className="page-heading page-heading-without-breadcrumb">
              <h1 className="au-display-xl">Reset password</h1>
            </header>
            <Form
              model={model}
              id="resetPassword"
              validators={{
                '': {
                  // Form-level validator
                  passwordsMatch: vals => vals.password === vals.confirmPassword
                }
              }}
              validateOn="submit"
              onSubmit={data => handleSubmit(data)}
            >
              <Textfield
                model={`${model}.password`}
                name="password"
                id="password"
                htmlFor="password"
                label="Password"
                type="password"
                description="Your password should be at least 10 characters"
                validators={{ passwordLength }}
                messages={{
                  passwordLength: 'Your password should be at least 10 characters'
                }}
              />
              <Textfield
                model={`${model}.confirmPassword`}
                name="confirmPassword"
                id="confirmPassword"
                htmlFor="confirmPassword"
                label="Password"
                type="password"
                description="Repeat password used above"
                validators={{
                  passwordLength
                }}
                messages={{
                  passwordLength: 'Your password confirmation should be at least 10 characters. ',
                  passwordsMatch: 'Passwords do not match.'
                }}
              />
              <p className={styles.buttonWrapper}>
                <input className="au-btn" type="submit" value="Reset password" onClick={submitClicked} />
              </p>
            </Form>
          </article>
        </div>
      </div>
    </main>
  )
}

ResetPasswordForm.defaultProps = {
  submitClicked: null,
  handleSubmit: null
}

ResetPasswordForm.propTypes = {
  user: PropTypes.shape({
    resetPasswordSuccess: PropTypes.bool
  }).isRequired,
  model: PropTypes.string.isRequired,
  submitClicked: PropTypes.func,
  handleSubmit: PropTypes.func
}

export default ResetPasswordForm
