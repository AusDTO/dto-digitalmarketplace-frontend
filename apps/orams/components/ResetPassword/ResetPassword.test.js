import React from 'react'
import expect from 'expect'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'

import configureStore from '../../store'
import RequestResetEmailForm from './RequestResetEmailForm'
import ResetPasswordForm from './ResetPasswordForm'

describe('Test suite for Reset Password Functional Components', () => {
  const state = {
    user: {
      resetPasswordSuccess: true,
      resetPasswordEmailSuccess: true
    },
    userRegistrationDetails: {
      name: 'Jeff Labowski',
      email_address: 'e@mail.com'
    },
    loadCompleteSuccess: true,
    loadRegistrationData: () => true,
    location: { pathname: '/2' },
    match: { url: '/url' },
    handleSubmit: () => true
  }

  const store = configureStore()
  test('RequestResetEmailForm renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <RequestResetEmailForm {...state} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  test('ResetPasswordForm renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <ResetPasswordForm {...state} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
