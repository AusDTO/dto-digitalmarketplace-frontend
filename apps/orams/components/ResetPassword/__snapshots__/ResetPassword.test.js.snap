// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Reset Password Functional Components RequestResetEmailForm renders without errors 1`] = `
<RequestResetEmailForm
  handleSubmit={[Function]}
  loadCompleteSuccess={true}
  loadRegistrationData={[Function]}
  location={
    Object {
      "pathname": "/2",
    }
  }
  match={
    Object {
      "url": "/url",
    }
  }
  user={
    Object {
      "resetPasswordEmailSuccess": true,
      "resetPasswordSuccess": true,
    }
  }
  userRegistrationDetails={
    Object {
      "email_address": "e@mail.com",
      "name": "Jeff Labowski",
    }
  }
/>
`;

exports[`Test suite for Reset Password Functional Components ResetPasswordForm renders without errors 1`] = `
<ResetPasswordForm
  handleSubmit={[Function]}
  loadCompleteSuccess={true}
  loadRegistrationData={[Function]}
  location={
    Object {
      "pathname": "/2",
    }
  }
  match={
    Object {
      "url": "/url",
    }
  }
  submitClicked={null}
  user={
    Object {
      "resetPasswordEmailSuccess": true,
      "resetPasswordSuccess": true,
    }
  }
  userRegistrationDetails={
    Object {
      "email_address": "e@mail.com",
      "name": "Jeff Labowski",
    }
  }
/>
`;
